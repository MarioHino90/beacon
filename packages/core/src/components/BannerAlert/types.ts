import React from "react";
import { ThemedStyledProps } from "styled-components";

import { UntransitionedBannerAlertProps } from "../UntransitionedBannerAlert";

export type Intent = "success" | "warning" | "danger" | "info";

export interface BannerAlertProps
  extends UntransitionedBannerAlertProps,
    Omit<StyledBannerTransitionProps, "children"> {}

export interface StyledBannerTransitionProps
  extends React.HTMLAttributes<HTMLDivElement> {
  isOpen: boolean;
}

export interface IntentContextProviderProps {
  intent: Intent;
}

// this uses <any, any> because the extra props in the themed styled component
// do not matter, just that it is themed
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface IntentColorArgs extends ThemedStyledProps<any, any> {
  intent: Intent;
}
