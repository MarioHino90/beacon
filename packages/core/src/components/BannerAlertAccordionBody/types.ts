export interface BannerAlertAccordionBodyProps
  extends React.HTMLAttributes<HTMLDivElement> {}
