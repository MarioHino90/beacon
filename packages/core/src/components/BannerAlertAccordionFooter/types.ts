export interface BannerAlertAccordionFooterProps
  extends React.HTMLAttributes<HTMLDivElement> {}
