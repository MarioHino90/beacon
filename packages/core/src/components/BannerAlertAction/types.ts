import { ButtonProps } from "../Button";
import { Intent } from "../BannerAlert";

export interface BannerAlertActionProps
  extends Omit<ButtonProps, "variant" | "intent"> {
  /** * the visual intent */
  intent?: Intent;
}
