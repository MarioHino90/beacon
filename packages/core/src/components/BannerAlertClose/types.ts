import { Intent } from "../BannerAlert";

export interface BannerAlertCloseProps {
  /**
   * visual banner intents */
  intent: Intent;
  /**
   * on close callack */
  onClose: () => void;
}

export interface StyledProps {
  /**
   * visual banner intents */
  intent: Intent;
}
