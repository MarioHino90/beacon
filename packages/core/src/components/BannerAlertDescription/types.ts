export interface BannerAlertDescriptionProps
  extends React.HTMLAttributes<HTMLDivElement> {}
