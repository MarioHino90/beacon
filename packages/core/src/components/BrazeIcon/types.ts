import React from "react";

export type GenericSpotIcon<Props> = React.LazyExoticComponent<{
  (props: Props): React.ReactElement;
  displayName: string;
}>;

export interface SpotIcons {
  "add-document": GenericSpotIcon<AddDocumentProps>;
  "add-image-or-document": GenericSpotIcon<AddImageOrDocumentProps>;
  "arrow-toggle": GenericSpotIcon<ArrowToggleProps>;
  "campaign-comparison": GenericSpotIcon<CampaignComparisonProps>;
  content: GenericSpotIcon<ContentProps>;
  "email-settings": GenericSpotIcon<EmailSettingsProps>;
  "empty-table": GenericSpotIcon<AddDocumentProps>;
  "funnel-report": GenericSpotIcon<FunnelReportProps>;
  mms: GenericSpotIcon<MMSProps>;
  preview: GenericSpotIcon<PreviewProps>;
  "retention-report": GenericSpotIcon<RetentionReportProps>;
  sms: GenericSpotIcon<SMSProps>;
}

export interface BrazeIconProps extends StyledBrazeIconProps {
  /**
   * icon to be displayed to user */
  name?: keyof SpotIcons;
}

export interface StyledBrazeIconProps
  extends React.HTMLAttributes<HTMLElement> {
  /**
   * child elements to be rendered inside the component tree */
  children?: React.ReactNode | React.ReactNode[];
}

export interface AddDocumentProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface AddImageOrDocumentProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface ArrowToggleProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface CampaignComparisonProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface ContentProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface FunnelReportProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface EmailSettingsProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface PreviewProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface MMSProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface RetentionReportProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface SMSProps extends React.SVGAttributes<HTMLOrSVGElement> {}
