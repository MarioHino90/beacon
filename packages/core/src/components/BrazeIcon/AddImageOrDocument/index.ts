export { AddImageOrDocument } from "./AddImageOrDocument";

export type { AddImageOrDocumentProps } from "../types";
