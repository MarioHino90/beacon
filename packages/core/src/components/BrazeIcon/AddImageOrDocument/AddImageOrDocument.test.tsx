import React from "react";
import { shallow } from "enzyme";
import faker from "@faker-js/faker";

import { AddImageOrDocumentProps } from "../types";

import { AddImageOrDocument } from "./AddImageOrDocument";

faker.seed(1604940599873);

describe("AddImageOrDocument", () => {
  let props: AddImageOrDocumentProps;

  beforeEach(() => {
    props = {
      children: faker.random.words(),
      className: faker.random.word(),
      id: faker.random.word(),
    };
  });

  it("should match the snapshot", () => {
    const wrapper = shallow(<AddImageOrDocument {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
