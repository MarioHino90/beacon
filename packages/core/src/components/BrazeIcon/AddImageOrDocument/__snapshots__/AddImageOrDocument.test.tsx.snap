// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddImageOrDocument should match the snapshot 1`] = `
<svg
  className="bcl-add-image-or-document Savings"
  fill="none"
  height="77"
  id="input"
  viewBox="0 0 75 77"
  width="75"
  xmlns="http://www.w3.org/2000/svg"
>
  <rect
    fill="white"
    height="38.1579"
    rx="4"
    width="38.1579"
    x="36.4476"
  />
  <rect
    fill="#75DBE7"
    height="26.3158"
    rx="2"
    width="26.3158"
    x="41.7107"
    y="6.5791"
  />
  <path
    d="M46.9739 26.3159H62.7634"
    stroke="#101B24"
    strokeWidth="2"
  />
  <rect
    height="37.4737"
    rx="3"
    stroke="#101B24"
    strokeWidth="2"
    width="37.4737"
    x="36.1318"
    y="1"
  />
  <rect
    fill="white"
    height="48.6842"
    rx="4"
    width="48.6842"
    x="4.86865"
    y="22.3687"
  />
  <path
    d="M24.6055 36.8423L52.625 69.737H10.3628C6.26812 69.737 3.90946 65.0844 6.32979 61.7816L24.6055 36.8423Z"
    fill="#C4F0F5"
  />
  <rect
    height="46.6842"
    rx="3"
    stroke="#101B24"
    strokeWidth="2"
    width="46.6842"
    x="5.86865"
    y="23.3687"
  />
  <circle
    cx="40.395"
    cy="34.2104"
    fill="#75DBE7"
    r="5.26316"
  />
  <circle
    cx="51.5791"
    cy="65.1315"
    fill="#75DBE7"
    r="10.1842"
    stroke="#101B24"
    strokeWidth="2"
  />
  <path
    d="M51.7434 59.2104V70.7236"
    stroke="#101B24"
    strokeWidth="2"
  />
  <path
    d="M57.5 64.9668L45.9868 64.9668"
    stroke="#101B24"
    strokeWidth="2"
  />
  <path
    d="M6.18445 60.3131L25.1903 36.8423L43.0266 58.5528"
    stroke="#101B24"
    strokeWidth="2"
  />
  <path
    d="M46.9739 13.1577H62.7634"
    stroke="#101B24"
    strokeWidth="2"
  />
  <path
    d="M46.9739 19.7368H62.7634"
    stroke="#101B24"
    strokeWidth="2"
  />
</svg>
`;
